lib/workos/configuration.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+# typed: false
+
+module WorkOS
+  # Configuration class sets config initializer
+  class Configuration
+    attr_accessor :timeout
```

Comment:
Gotta love configuration blocks!

If we are going to start using this pattern, seems like we should consolidate other configuration here as well, right? Thankfully, seems like that would only mean `key` for the moment.

If we don't, developers who configure both `key` and `timeout` would need to use two different patterns:

```rb
# Not great

WorkOS.key = 'key_123abc'
WorkOS.configure do |config|
  config.timeout = 120
end

# Better

WorkOS.configure do |config|
  config.key = 'key_123abc'
  config.timeout = 120
end
```

We'd probably also want to remove `Base` in favor of `Configuration`, and deprecate the key-related singleton class methods on the `WorkOS` module (`key`, `key=`, `key!`, etc.). We need to keep them around for backwards compatibility, but we can implement them in terms of the new pattern:

```rb
module WorkOS
  # ...

  def self.key=(value)
    warn '`WorkOS.key=` is deprecated. Use `WorkOS.configure` instead.'

    config.key = value
  end

  def self.key
    warn '`WorkOS.key` is deprecated. Use `WorkOS.config.key` instead.'

    config.key
  end

  # ...
end
```

@blairlunceford What do you think?

Reply:
We could also leave `WorkOS.key` as a shorthand for developers that don't want to set any other options, but it would still be good to define it in terms of `config`, so that there's a single source of truth for configuration in the Gem.

Reply:
Thanks! I moved the key to the configuration, and added the deprecation notices.